pandas/tests/frame/test_sorting.py

```diff
@@ -598,3 +598,71 @@ def test_sort_index_intervalindex(self):
             closed='right')
         result = result.columns.levels[1].categories
         tm.assert_index_equal(result, expected)
+
+    def test_sort_index_na_position_with_categories(self):
+        # GH 22556
+        # Positioning missing value properly when column is Categorical.
+        categories = ['A', 'B', 'C']
+        list_of_nans = [np.nan, np.nan]
+        df = pd.DataFrame({
+            'c': pd.Categorical(['A', np.nan, 'B', np.nan, 'C'],
+                                categories=categories,
+                                ordered=True)})
+        # sort ascending with na first
+        result = df.sort_values(by='c',
+                                ascending=True,
+                                na_position='first')
+        expected = DataFrame({
+            'c': Categorical(list_of_nans + categories,
+                             categories=categories,
+                             ordered=True)}, index=[1, 3, 0, 2, 4])
+
+        assert_frame_equal(result,
```

Comment:
Readability nit but can you make this statement one line (applicable in a few instances below)